pkg/cloud/cloud.go

```diff
@@ -653,7 +656,11 @@ func (c *cloud) ResizeOrModifyDisk(ctx context.Context, volumeID string, newSize
 
 	response, err := c.ec2.ModifyVolumeWithContext(ctx, req)
 	if err != nil {
-		return 0, fmt.Errorf("unable to modify AWS volume %q: %w", volumeID, err)
+		if isAWSErrorModifyVolumeInvalidParameter(err) {
+			// Wrap the many possible EC2 errors with cloud error, but pass on actual EC2 error to operator
+			return 0, fmt.Errorf("%w: %w", ErrInvalidArgument, err)
```

Comment:
Instead of wrapping the EC2 error with `ErrInvalidArgument`, could this return an `INVALID_ARGUMENT` gRPC error code directly?

```
return status.Error(codes.InvalidArgument, ...)
```

Reply:
Today `cloud` does not know about gRPC error codes. While this saves a wrap, suddenly `controller` pkg cannot trust whether or not RPC errors or cloud errors are being returned. 

I am open to this change if we are open to refactoring `cloud`, but I don't want modifyVolume to be an exception to the rule.